assets/src/components/editing/elements/table/TableDropdownMenu.tsx

```diff
@@ -1,24 +1,38 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useCallback } from 'react';
 import { ReactEditor } from 'slate-react';
 import { Transforms, Editor, Path, Element } from 'slate';
-import { TableData, TableHeader, TableRow } from 'data/content/model/elements/types';
+import {
+  Table,
+  TableBorderStyle,
+  TableCell,
+  TableData,
+  TableHeader,
+  TableRow,
+  TableRowStyle,
+} from 'data/content/model/elements/types';
 import { Model } from 'data/content/model/elements/factories';
+import {
+  canExpandCellRight,
+  canExpandDown,
+  canSplitCell,
+  expandCellDown,
+  expandCellRight,
+  splitCell,
+} from './table-cell-merge-operations';
 
 // Dropdown menu that appears in each table cell.
 interface Props {
   editor: Editor;
-  model: TableData | TableHeader;
+  model: TableCell;
 }
-export const DropdownMenu = (props: Props) => {
-  const ref = useRef();
-  const { editor, model } = props;
-
-  // There has to be a better way to do this.
```

Comment:
IMO it's better to use a reference function instead of a reference + useEffect for stuff like this. Should be safer and result in one less render call.